### PR TITLE
Add PHPCS & related Composer CLI scripts

### PR DESCRIPTION
**Pull request by ravage84**
Inspired by:
https://github.com/thephpleague/skeleton/blob/03e1f7e71c1a7257f462b253d5eb4dbf2c2403d2/composer.json

### CODE REVIEW

**Comment by raphaelstolt** [comment]
Can the Composer script `cs-check` be utilised on Travis CI builds? Does it exit with a non zero exit code when violations are detected? If so it would be nice to have a **single** check against one PHP version like this [one](https://github.com/jonathantorres/construct/blob/master/.travis.yml#L32).

**Comment by ravage84** [comment]
@raphaelstolt no, it cannot, as the cli parameters ensure that it always exists with a 0.

https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-errors-when-generating-the-exit-code

This was added by myself to prevent Composer misbehave, when phpcs finds errors.
When executing ``composer cs-check``, it should not "break" the execution, if it finds cs errors as the execution did not fail.

**Comment by raphaelstolt** [comment]
IMHO than the utilisation of PHP Coding Standards Fixer is more suitable also in regard to #25.

**Comment by ravage84** [comment]
@raphaelstolt  And why is it more suitable?

By the way, the two added code style related Composer scripts are primarily there for manual execution by the dev. Just for convenience.

>  If so it would be nice to have a single check against one PHP version like this one.

I guess, that's a topic for  [new issue](https://github.com/povils/phpmnd/issues/new)?

**Comment by raphaelstolt** [comment]
It's more suitable because it breaks the build early when a coding standard violation is introduced.

**Comment by ravage84** [comment]
I do not understand what the choice of tool has to do with the Travis build and how you configure the Travis build process.
With both tools, you should be able to let the Travis builds break as early as you want it to (though I personally question that intention).

Just to clarify again:

The introduced Composer scripts ``cs-check`` and ``cs-fix`` are for conventient, **manual** execution.
Normally, phpcs exits with a non-zero code, too, if it finds CS violations. See [the related documentation](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#ignoring-errors-when-generating-the-exit-code).

> By default, PHP_CodeSniffer will exit with a non-zero code if any errors or warnings are found.

**Comment by raphaelstolt** [comment]
Ok so the current configuration/call (in this PR) of `cs-check` does return a zero exit code even if a violation is detected. Sorry misread your comment. So the choice of tool doesn't matter. 😶 

I get your point that the scripts are primarly meant for manual execution but from experience I can tell you that even the most disciplined developer will forget or ignore it. So an utilisation in Travis CI builds __should__ break the build and thereby raise the chances that the contributing developer will fix it's introduced violations.

**Comment by ravage84** [comment]
The Composer scripts are not meant as a replacement for their counterparts in a CI system.

They are there to give a developer one way (read: "script") to check for errors and another way to fix them, __before__ providing a PR, for example. If s/he forgets it, a CI could/should still lament it or even fix it, if it can.

The PR workflow in the contribution guide could look something like this:

1. Fork the code
2. Clone your fork
3. Work in a branch
4. Fix/improve the code
5. Check/fix your changes locally with ``composer cs-check`` and ``composer cs-fix`` until it shows no cs violations
6. Commit & push
7. Open a PR
8. (automatic) Some CI system checks your changes against errors, maintainability and cs violations.
9. ...

If point 5 gets left behind, the cs violations can still get caught in point 8.

But providing those two proposed scripts we give developers some conventient tool to provide a quality PR as soon and as easy as possible.

**Comment by raphaelstolt** [comment]
> The Composer scripts are not meant as a replacement for their counterparts in a CI system.

Why not and what's the counterpart in a CI system to check coding standard violations? Why should it not be allowed to utilise tooling that works locally also in a CI system? Got that running on several projects with success.

**Comment by ravage84** [comment]
> Why should it not be allowed to utilise tooling that works locally also in a CI system?

Oh, have I said so? The Composer scripts are just like __shortcuts__ to the ``phpcs`` and ``phpcbf`` **tools**. And certainly, the CI can use those tools. It just uses them through a **non-shortcut** way.

The shortcuts are simplified ways of executing the cs check/fix in a very particular way. The CI system on the other hand does not necessarily need it the same way. I do not see the need for the CI to use one common command (the shortcut). It does not provide much advantage. To the contrary, it might limit the execution or makes the command behind the shortcut more complicated to maintain.

And the reason for returning a zero exit code is this:

https://github.com/thephpleague/skeleton/pull/41#issuecomment-104071697

If that issue does not exist anymore, we might not need the additional cli parameters...

**Comment by raphaelstolt** [comment]
Why should CI systems use them through a __non-shortcut__ way? If a CI specific _shortcut_ is needed a specific Composer script can be configured/defined.

For me Composer scripts are tools/ways to wrap or hide (configuration) complexity to a certain degree. If it gets too complex they get extracted into bash scripts or the like anyways; but that's not the case for coding standard checks and test execution. If these work locally they will also work on the CI system.

**Comment by ravage84** [comment]
> Why should CI systems use them through a non-shortcut way?

Simply, because shortcuts have two main reasons for existance:

Something gets either used very often manually (a) or when it get used, it takes long/is complicated (b).

Since the execution of the cs check is programmed once, executed automatically many times and thus neither a nor b is relevant, I do not feel the need to add a shortcut for the CI.

>  If a CI specific shortcut is needed a specific Composer script can be configured/defined.

A shortcut is not _needed, but sure, you can add as many Composer scripts you feel the need in your own PR. This is not my project, but this is my PR.

> For me Composer scripts are tools/ways to wrap or hide (configuration) complexity to a certain degree. 

Certainly, a possible use for them. I normally prefer configuration files, like phpunit.xml over Composer scripts.

> If these work locally they will also work on the CI system.

Generally, yes.

To sum it up for me:
I feel my PR is "feature complete" as adding any Travis CI related stuff is not part of my PR.
Also because I still stand behind my proposal #25 to use a CS focused CI , like Stickler CI  .

**Comment by raphaelstolt** [comment]
👍 — My roll of 2¢s is empty, anyways.

**Comment by povils** [comment]
Dont see big advantage of that, but dont see any harm also
